test/box-luatest/downgrade_helper.lua

```diff
@@ -0,0 +1,71 @@
+local t = require('luatest')
+
+-- Parse string with version in format 'A.B.C' to version id.
+local function parse_version(version)
+    local major, minor, patch = version:match('^(%d+)%.(%d+)%.(%d+)$')
+    if major == nil then
+        error('version should be in format A.B.C')
+    end
+    return bit.bor(bit.lshift(bit.bor(bit.lshift(major, 8), minor), 8), patch)
+end
+
+-- Find oldest application version that can run on given schema version.
+-- Return index of this application version in box.schema.downgrade_versions.
+local function app_version_index(version_str)
+    local version = parse_version(version_str)
+    local versions = box.schema.downgrade_versions()
+    local idx
+    for i, v_str in ipairs(versions) do
+        local v = parse_version(v_str)
+        if v >= version then
+            idx = i
+            break
+        end
+    end
+    return idx
+end
```

Comment:
Nit: perhaps it would be better to export these function from upgrade script via `box.internal` instead of copy-pasting their implementations?

Reply:
But there is only one such function - `parse_version`.

@alyapunov  what do you think of such exposing?